src/main/js/helpers/axis.js

```diff
@@ -191,37 +191,136 @@ const getAxesScale = (axis, scale, config) => {
         getAxisInfoOrientation(config.axis.x.orientation)
     );
 
-    // If ticksCount is undefined or greater than TICKS_MAXCOUNT
-    // AND if the Y2 is visible, then utilize a default value for
-    // the ticksCount. This is based on the ranges of the Y & Y2 axes.
+    if (hasY2Axis(config.axis)) {
+        if (!utils.isUndefined(config.axis.y.ticks.values)) {
+            axis.y = prepareYAxis(
+                scale.y,
+                config.axis.y.ticks.values,
+                config.height,
+                getAxisTickFormat(config.locale, config.axis.y.ticks.format)
+            );
+            axis.y2 = prepareY2Axis(
+                scale.y2,
+                config.axis.y2.ticks.values,
+                config.height,
+                getAxisTickFormat(config.locale, config.axis.y2.ticks.format)
+            );
+            return axis;
+        }
```

Comment:
We may update the logic like this

```js
// If y-2 axis is present
if (hasY2Axis(config.axis)) {
     // If tick values are provided, ticks calculated based on provided values
     if (utils.isDefined(config.axis.y.ticks.values) && utils.isDefined(config.axis.y2.ticks.values)) {
     ......................................................
    } 
    // If ticksCount property is provided and it is less than the maximum ticks count
   // tick values calculated based on the ticksCount
   else if (utils.isDefined(config.ticksCount) && config.ticksCount <= constants.TICKSCOUNT_MAXLIMIT) {
     ......................................
   }
      // If ticksCount is undefined or greater than TICKS_MAXCOUNT
      // AND if the Y2 is visible, then utilize a default value for
     // the ticksCount. This is based on the ranges of the Y & Y2 axes.
    else {
      .............................
     }

// If y-2 axis not present and only y-axis present
else {
     // If y axis tick values are provided, ticks calculated based on provided values
     if (utils.isDefined(config.axis.y.ticks.values)) {
     ......................................................
    } 
    // If y-axis ticksCount property is provided and it is less than the maximum ticks count
   // tick values calculated based on the ticksCount
   else if (utils.isDefined(config.ticksCount) && config.ticksCount <= constants.TICKSCOUNT_MAXLIMIT) {
     ......................................
   }
      // If ticksCount is undefined or greater than TICKS_MAXCOUNT
      //  then utilize a default value for
     // the ticksCount. This is based on the ranges of the Y axis.
    else {
      .............................
     }
} 
```

Reply:
Is there any advantage to updating the logic this way? The behavior is identical to mine, but as a developer, I would prefer to use only `if` and multiple returns, because I can see that once the condition is true, the function ends, so no need to continue reading if I'm debugging a certain condition.

Reply:
It is not necessary to follow any particular way. But we should try to write a code which is more readable. And the second thing is if it is known that if only one condition can be true at a time then it is better to use the if-else ladder as it will save time for unnecessary checking of all if the condition

Reply:
update here:
https://github.com/cerner/carbon-graphs/pull/249/commits/a805317e11652ca6ea8f9c4dc607f86d9e2caf33